op-node/node/node.go

```diff
@@ -132,136 +152,214 @@ func New(ctx context.Context, cfg *config.Config, log log.Logger, appVersion str
 	return n, nil
 }
 
-func (n *OpNode) init(ctx context.Context, cfg *config.Config) error {
+type initializationOverrides struct {
+	L1Source   L1Source
+	Beacon     L1Beacon
+	RPCHandler *oprpc.Handler
+}
+
+// init progressively creates and sets up all the components of the OpNode
+// some later initialization steps depend on the node being partially initialized with other components,
+// so order is important to ensure that all resources are available when needed.
+func (n *OpNode) init(ctx context.Context, cfg *config.Config, overrides *initializationOverrides) error {
 	n.log.Info("Initializing rollup node", "version", n.appVersion)
-	n.initEventSystem()
-	if err := n.initL1BeaconAPI(ctx, cfg); err != nil {
-		return err
+	if overrides == nil {
+		overrides = &initializationOverrides{}
```

Comment:
I'm not sure if this is a Go anti-pattern but I'd be very tempted to just have a non-pointer `initiatlizationOverrides` since all of its components are pointer types anyway. Then we wouldn't need to handle the nil case here.